opts.go

```diff
@@ -0,0 +1,80 @@
+package txfile
+
+import "fmt"
+
+// Options provides common file options used when opening or creating a file.
+type Options struct {
+	// Additional flags.
+	Flags Flag
+
+	// MaxSize sets the maximum file size in bytes. This should be a multiple of PageSize.
+	// If it's not a multiple of PageSize, the actual files maximum size is rounded downwards
+	// to the next multiple of PageSize.
+	// A value of 0 indicates the file can grow without limits.
+	MaxSize uint64
+
+	// PageSize sets the files page size on file creation. PageSize is ignored if
+	// the file already exists.
+	// If PageSize is not configured, the OSes main memory page size is selected.
+	PageSize uint32
+
+	// InitMetaArea configures the minimum amount of page in the meta area.
+	// The amount of pages is only allocated when the file is generated.
+	// The meta area grows by double the current meta area. To reduce the
+	// total amount of pages moved to meta area on grow, it is recommended that
+	// the value of InitMetaArea is a power of 2.
```

Comment:
To be honest I think we should enforce that? WDYT?

Reply:
I didn't want to enforce it. E.g. assuming we don't have much fragmentation, the `pq` will run fine with 10 pages being preallocated. Even when growing, less pages might be used in the end, then starting with 16. Instead I will add a grow factor that must be > 1 in the future.

Reply:
Sound good!